File: AdventJS/2025/reto 6/6.py
gloves = [
  { "hand": 'L', "color": 'red' },
  { "hand": 'R', "color": 'red' },
  { "hand": 'R', "color": 'green' },
  { "hand": 'L', "color": 'blue' },
  { "hand": 'L', "color": 'green' }
]

counter = {}
pares = []

for glove in gloves:
  counter.setdefault(glove["color"], 0)

  if glove["hand"] == "L" and counter[glove["color"]] < 0 or \
     glove["hand"] == "R" and counter[glove["color"]] > 0:
     pares.append(glove["color"])

  counter[glove["color"]] += 1 if glove["hand"] == "L" else -1

print(pares)




from typing import List, Dict

def match_gloves(gloves: List[Dict[str, str]]) -> List[str]:
  counter = {}
  pares = []

  for glove in gloves:
    counter.setdefault(glove["color"], 0)

    if glove["hand"] == "L" and counter[glove["color"]] < 0 or \
       glove["hand"] == "R" and counter[glove["color"]] > 0:
       pares.append(glove["color"])

    counter[glove["color"]] += 1 if glove["hand"] == "L" else -1
  return pares
  

File: AdventJS/2025/reto 6/6.js
/**
 * @param {{ hand: 'L' | 'R', color: string }[]} gloves
 * @returns {string[]} Colors of matched pairs
 */
function matchGloves(gloves) {
  const counter = {};
  const pares = [];

  for (const { hand, color } of gloves) {
    counter[color] ??= 0;
    
    if ((hand === "L" && counter[color] < 0) ||
        (hand === "R" && counter[color] > 0)) {
      pares.push(color);
    }

    counter[color] += hand === "L" ? 1 : -1;
  }

  return pares;
}

// Usar ??= para inicializar contadores.